resources/net.go

```diff
@@ -0,0 +1,726 @@
+// Mgmt
+// Copyright (C) 2013-2017+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/purpleidea/mgmt/recwatch"
+	"github.com/purpleidea/mgmt/util"
+
+	systemd "github.com/coreos/go-systemd/dbus" // change namespace
+	systemdUtil "github.com/coreos/go-systemd/util"
+	"github.com/godbus/dbus"
+	errwrap "github.com/pkg/errors"
+)
+
+const (
+	// IfacePrefix is used to identify unit files for managed links.
+	IfacePrefix = "mgmt-"
+	// IfaceUp is the up interface state
+	IfaceUp = "up"
+	// IfaceDown is the down interface state
+	IfaceDown = "down"
+
+	// DBusInterface is DBus's own DBus interface.
+	DBusInterface = "org.freedesktop.DBus"
+	// DBusAddMatch is the DBus method to filter signals.
+	DBusAddMatch = DBusInterface + ".AddMatch"
+	// DBusPropertiesInterface is the DBus interface used to get events.
+	DBusPropertiesInterface = DBusInterface + ".Properties"
+	// DBusPropertiesGet is the DBus method to get a property.
+	DBusPropertiesGet = DBusPropertiesInterface + ".Get"
+	// DBusNetworkdInterface is the networkd dbus interface.
+	DBusNetworkdInterface = "org.freedesktop.network1"
+	// DBusNetworkdLinkInterface is the networkd dbus link interface.
+	DBusNetworkdLinkInterface = DBusNetworkdInterface + ".Link"
+	// DBusLinkPath is the DBus source path for network interface events.
+	DBusLinkPath = "/org/freedesktop/network1/link/"
+	// DBusEavesdropSprint is the format of dbus eavesdrop arguments.
+	DBusEavesdropSprint = "type='signal', interface='%s', path='%s', eavesdrop='true'"
+	// DBusFailMode is the dbus mode used for restarting the network service
+	DBusFailMode = "fail"
+
+	// NetworkdUnitFileDir is the location of networkd unit files.
+	NetworkdUnitFileDir = "/etc/systemd/network/"
+	// NetworkdUnitFileExt is the file extension for networkd unit files.
+	NetworkdUnitFileExt = ".network"
+	// NetworkdService is the name of the networkd service.
+	NetworkdService = "systemd-networkd.service"
+
+	// ActiveState is the systemd-networkd service's active state.
+	ActiveState = "ActiveState"
+	// ActiveStateActive is the active service state.
```

Comment:
These and a few other comments on these consts are in the form: `foo is the foo of the whatever`. While I've certainly been guilt of this kind of thing before, if you could update a few of them so that they're a little bit more self descriptive for those that don't know what the internal networkd API is all about. Remember, these comments get displayed in godoc.

The fact that you have all these const's is probably good! Let's just make them useful in the docs if someone would ever need them, or private if they should never be used externally.

Why should they be public or private? If a third party wants to predict what we might be doing, knowing the prefix (eg `mgmt-`) is something we need public. Also if a consumer of this resource when using mgmt as a lib wants to know something that it should be able to gleam from this, that's useful too.